assets/javascripts/discourse/controllers/preferences-saved-searches.js

```diff
@@ -0,0 +1,50 @@
+import Controller from "@ember/controller";
+import { action } from "@ember/object";
+import discourseComputed from "discourse-common/utils/decorators";
+import { ajax } from "discourse/lib/ajax";
+import { popupAjaxError } from "discourse/lib/ajax-error";
+
+export default Controller.extend({
+  saving: false,
+
+  @discourseComputed("savedSearches.length", "siteSettings.max_saved_searches")
+  canAddSavedSearch(length, max) {
+    return length < max;
+  },
+
+  @action
+  addSavedSearch() {
+    this.savedSearches.pushObject({ query: "" });
+  },
+
+  @action
+  removeSavedSearch(savedSearch) {
+    this.savedSearches.removeObject(savedSearch);
+  },
+
+  @action
+  save() {
+    this.setProperties({ saved: false, isSaving: true });
+
+    const savedSearches = this.savedSearches
+      .map((savedSearch) => {
+        const query = savedSearch.query.trim();
+        return query ? query : null;
+      })
+      .compact();
+
+    return ajax(`/u/${this.model.username}/preferences/saved-searches`, {
+      type: "PUT",
+      dataType: "json",
+      data: { searches: savedSearches },
+    })
+      .then(() =>
+        this.setProperties({
+          saved: true,
+          "model.saved_searches": savedSearches,
+        })
+      )
+      .catch(popupAjaxError)
+      .finally(() => this.set("isSaving", false));
```

Comment:
you declared "saving" and not "isSaving" at the top of the file

Reply:
btw where is this used? Sorry if I missed it, I don't have the project locally.

Reply:
This is used in `{{save-controls}}`.

Reply:
Ok we should probably change save-controls default to have `saved: false,` then

Reply:
This component is passed to that component, so I think it makes sense to keep `saved: false` here.